src/qibolab/platforms/multiqubit.py

```diff
@@ -28,10 +92,31 @@ def execute_pulse_sequence(self, sequence: PulseSequence, nshots=None):
         # Process Pulse Sequence. Assign pulses to instruments and generate waveforms & program
         instrument_pulses = {}
         roles = {}
+
+        changed = {}
         for name in self.instruments:
             roles[name] = self.settings["instruments"][name]["roles"]
             if "control" in roles[name] or "readout" in roles[name]:
                 instrument_pulses[name] = sequence.get_channel_pulses(*self.instruments[name].channels)
+                # Change pulses frequency to if and correct lo accordingly (before was done in qibolab)
+
+                if "readout" in roles[name]:
```

Comment:
These `"readout"` and `"control"` are another good candidate for `enum`, since a role should not be any possible string, but just one of those options.

Reply:
Of course, in order to use an `enum`, it should be an attribute of `roles[name]`, that in any case it would be better style than checking a substring
```suggestion
                if roles[name].is_readout():
```
or
```suggestion
                if Role.readout in roles[name]:
```
making `Role` a subclass of [`Flag`](https://docs.python.org/3/library/enum.html#enum.Flag).